### PR TITLE
✨ Implement blank handlers

### PR DESCRIPTION
**Pull request by ElRaptorus**
**Changes:**

1. Add types for `EmptyActivities`.
2. Add new REST Settings, Socket Settings and message types for handling notifications related to `EmptyActivities`.
3. Add interface for `IEmptyActivityConsumerAPI`
4. Refactor the `Message` namespace
    - Remove all `internal` message types, because they are not needed.
    - Move all messages contained in the `Messages.Public` namespace one level up and remove the `Public` sub-namespace. 

**Issues:**

Part of https://github.com/process-engine/process_engine_runtime/issues/270

PR: #43

## How can others test the changes?

Use the implemented contracts to parse and execute empty Activities.

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: src/apis/iempty_activity_management_api.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
+import {Subscription} from '@essential-projects/event_aggregator_contracts';
+import {IIdentity} from '@essential-projects/iam_contracts';
+
+import {EmptyActivityList} from '../data_models/empty_activity/index';
+import {Messages} from '../messages/index';
+
+/**
+ * The IEmptyActivityManagementApi is used to retreive and manage EmptyActivities.
```
Comment (Vyperus) [inline src/apis/iempty_activity_management_api.ts]:
```suggestion
 * The IEmptyActivityManagementApi is used to retrieve and manage EmptyActivities.
```

File: src/apis/iempty_activity_management_api.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+   */
+  finishEmptyActivity(
+    identity: IIdentity,
+    processInstanceId: string,
+    correlationId: string,
+    emptyActivityInstanceId: string,
+  ): Promise<void>;
+
+  /**
+   * Executes a callback when a EmptyActivity is reached.
```
Comment (Vyperus) [inline src/apis/iempty_activity_management_api.ts]:
```suggestion
   * Executes a callback when an EmptyActivity is reached.
```

File: src/apis/iempty_activity_management_api.ts [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+   * @returns               The Subscription created by the EventAggregator.
+   */
+  onEmptyActivityWaiting(
+    identity: IIdentity,
+    callback: Messages.CallbackTypes.OnEmptyActivityWaitingCallback,
+    subscribeOnce?: boolean,
+  ): Promise<Subscription>;
+
+  /**
+   * Executes a callback when a EmptyActivity is finished.
```
Comment (Vyperus) [inline src/apis/iempty_activity_management_api.ts]:
```suggestion
   * Executes a callback when an EmptyActivity is finished.
```

File: src/apis/iempty_activity_management_api.ts [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+   * @returns               The Subscription created by the EventAggregator.
+   */
+  onEmptyActivityFinished(
+    identity: IIdentity,
+    callback: Messages.CallbackTypes.OnEmptyActivityFinishedCallback,
+    subscribeOnce?: boolean,
+  ): Promise<Subscription>;
+
+  /**
+   * Executes a callback when a EmptyActivity for the given identity is reached.
```
Comment (Vyperus) [inline src/apis/iempty_activity_management_api.ts]:
```suggestion
   * Executes a callback when an EmptyActivity for the given identity is reached.
```

File: src/apis/iempty_activity_management_api.ts [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+   * Gets all waiting EmptyActivities belonging to the given identity.
+   *
+   * @async
+   * @param   identity The identity for which to get the EmptyActivities.
+   * @returns          The list of EmptyActivities.
+   */
+  getWaitingEmptyActivitiesByIdentity(identity: IIdentity): Promise<EmptyActivityList>;
+
+  /**
+   * Finishes a EmptyActivity belonging to an instance of a specific ProcessModel
```
Comment (Vyperus) [inline src/apis/iempty_activity_management_api.ts]:
```suggestion
   * Finishes an EmptyActivity belonging to an instance of a specific ProcessModel
```

File: src/messages/system_events/empty_activity_finished.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import {BaseEventMessage} from '../base_event_message';
+
+/**
+ * Represents the message which is send, when a EmptyActivity has been finished.
```
Comment (Vyperus) [inline src/messages/system_events/empty_activity_finished.ts]:
```suggestion
 * Represents the message which is send, when an EmptyActivity has been finished.
```

File: src/messages/system_events/empty_activity_reached.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import {BaseEventMessage} from '../base_event_message';
+
+/**
+ * Represents the message which is send, when a EmptyActivity has been finished.
```
Comment (Vyperus) [inline src/messages/system_events/empty_activity_reached.ts]:
```suggestion
 * Represents the message which is send, when an EmptyActivity has been finished.
```

File: src/messages/system_events/user_task_finished.ts [hunk 8]
```diff
@@ -1,14 +1,14 @@
-import {UserTaskResult} from '../../../data_models/user_task/user_task_result';
+import {UserTaskResult} from '../../data_models/user_task/user_task_result';
 
-import {BasePublicEventMessage} from '../base_public_event_message';
+import {BaseEventMessage} from '../base_event_message';
 
 /**
- * The message sent when a user task has been finished.
+ * The message sent when a UserTask has been finished.
```
Comment (Vyperus) [inline src/messages/system_events/user_task_finished.ts]:
```suggestion
 * The message sent when an UserTask has been finished.
```

File: src/messages/system_events/user_task_reached.ts [hunk 9]
```diff
@@ -1,14 +1,14 @@
-import {UserTaskResult} from '../../../data_models/user_task/user_task_result';
+import {UserTaskResult} from '../../data_models/user_task/user_task_result';
 
-import {BasePublicEventMessage} from '../base_public_event_message';
+import {BaseEventMessage} from '../base_event_message';
 
 /**
- * The message sent when a user task has been finished.
+ * The message sent when a UserTask has been finished.
```
Comment (Vyperus) [inline src/messages/system_events/user_task_reached.ts]:
```suggestion
 * The message sent when an UserTask has been finished.
```

File: src/messages/system_events/user_task_finished.ts (hunk 8 above)
Comment (ElRaptorus) [reply src/messages/system_events/user_task_finished.ts]: Nope.
no "an" before a "User".

See:
https://english.stackexchange.com/questions/105116/is-it-a-user-or-an-user

File: src/messages/system_events/user_task_reached.ts (hunk 9 above)
Comment (ElRaptorus) [reply src/messages/system_events/user_task_reached.ts]: See above.

**Pull request merged**